vulkaninfo/vulkaninfo.h

```diff
@@ -503,7 +503,8 @@ static void AppCreateXcbWindow(AppInstance &inst) {
     if (conn_error) {
         fprintf(stderr, "XCB failed to connect to the X server due to error:%d.\n", conn_error);
         fflush(stderr);
-        inst.xcb_connection = nullptr;
```

Comment:
I see that `xcb_disconnect(inst.xcb_connection);` needs to be called for freeing memory, but I fail to see why the nulling out of the inst.xcb_connection should be removed. It serves as a guard for later code to insure nothing uses the invalid xcb_connection.

Reply:
On line 500 above it says API guarantees non-null xcb_connection, shouldn't that be taken care of elsewhere?

Reply:
The comment refers to xcb_connect() never returning a null pointer, so unless we also kept the conn_error around, the only way to know if it is valid is to null it out when it fails. Code later on, namely AppCreateXcbSurface, need to know that and thus checks if xcb_connection is null.
xcb_disconnect() doesn't null out the pointer (since its pass-by-value), so we need to do it ourselves.

Reply:
ok, I can restore it I would like to confirm that nothing else will assume xcb_connection is not null like xcb_get_setup did.

Reply:
That should be good.

Reply:
And yes, AppCreateXcbSurface is the only function that is affected (that mattes) where the nullptr is needed for correct behavior.

Reply:
I pushed an updated version, let me know if something else has to be addressed.